src/Countries/Morocco.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+namespace Spatie\Holidays\Countries;
+
+use Carbon\CarbonImmutable;
+use Spatie\Holidays\Concerns\Translatable;
+use Spatie\Holidays\Contracts\HasTranslations;
+use Spatie\Holidays\Exceptions\InvalidYear;
+
+class Morocco extends Country implements HasTranslations
+{
+    use Translatable;
+
+    public function countryCode(): string
+    {
+        return 'ma';
+    }
+
+    public function defaultLocale(): string
+    {
+        return 'en';
+    }
+
+    protected function allHolidays(int $year): array
+    {
+        return array_merge([
+            'New Year\'s Day' => '01-01',
+            'Proclamation of Independence Day' => '01-11',
+            'Amazigh New Year (ⵉⴹ ⵏ ⵢⵉⵏⵏⴰⵢⵔ)' => '01-14',
+            'Labour Day' => '05-01',
+            'Throne Day' => '07-30',
+            'Oued Ed-Dahab Day' => '08-14',
+            'Revolution Day' => '08-20',
+            'Youth Day' => '08-21',
+            'Green March' => '11-06',
+            'Independence Day' => '11-18',
+        ], $this->variableHolidays($year));
+    }
+
+    /** @return array<string, CarbonImmutable> */
+    protected function variableHolidays(int $year): array
+    {
+        // Calculate the current Hijri year based on the Gregorian year
```

Comment:
You can have a look at other Islamic countries and how their lunar based holidays are resolved. For example: Turkey, Bahrein and Egypt.

Reply:
I've reviewed the other contributions, and it seems they've hardcoded the expected dates of Islamic holidays. Personally, I don't find that approach ideal. Instead, I opted to use a date converter for more dynamism. While it may not be 100% accurate, it offers flexibility. If you still prefer the hardcoded holiday expectation dates, I'm willing to make the change tonight.